contexts/AuthContext.js

```diff
@@ -0,0 +1,42 @@
+import React, { useContext, useState, useEffect } from 'react';
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  onAuthStateChanged,
+  signOut,
+} from 'firebase/auth';
+
+import { auth } from '../firebase/initFirebase';
+
+export const AuthContext = React.createContext();
+
+export const useAuth = () => useContext(AuthContext);
+
+export const AuthProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState();
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (user) => {
+      setCurrentUser(user);
+    });
+
+    return unsubscribe;
+  }, []);
+
+  const signup = (email, passwd) =>
+    createUserWithEmailAndPassword(auth, email, passwd);
+
+  const login = (email, passwd) =>
+    signInWithEmailAndPassword(auth, email, passwd);
+
+  const logout = () => signOut(auth);
+
+  const value = {
+    currentUser,
+    signup,
+    login,
+    logout,
+  };
```

Comment:
```suggestion
  const signup = useCallback((email, passwd) =>
    createUserWithEmailAndPassword(auth, email, passed), [auth]);

  const login = useCallback((email, passwd) =>
    signInWithEmailAndPassword(auth, email, passwd), [auth]);

  const logout = useCallback(() => signOut(auth), [auth]);

  const value = useMemo(() => ({
    currentUser,
    signup,
    login,
    logout,
  }), [ currentUser,
    signup,
    login,
    logout]);
```

Let's memorize the `value`. So it is not unnecessary to create a new object on every render. It could cause some unwanted rerenders